subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecHandle.java

```diff
@@ -260,7 +260,8 @@ public ExecHandle start() {
                 try {
                     stateChanged.await();
                 } catch (InterruptedException e) {
-                    //ok, wrapping up
+                    execHandleRunner.abortProcess();
+                    throw UncheckedException.throwAsUncheckedException(e);
```

Comment:
I think the fact that we interrupt the thread here should be explicit instead of hidden inside `throwAsUncheckedException()`.

Reply:
Same reason as above.